clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/BasicAmazonDeployHandler.groovy

```diff
@@ -241,7 +241,7 @@ class BasicAmazonDeployHandler implements DeployHandler<BasicAmazonDeployDescrip
       }
 
       if (description.useAmiBlockDeviceMappings) {
```

Comment:
Looks like we are setting description.blockDevices in two areas. Once here, and once on line 226. Does it make sense to have one method or area that the logic to set blockDevices can be constrained? It looks like there could be a scenario where a user could have blockDevices in the description and also have useAmiBlockDevice selected. This could be problematic. If it should only be 1 or the other, maybe we throw an error?

Reply:
For now I just added a logic to assign the **blockDevices** if the **useAmiBlockDevice** is false then check if the **blockDevices** is null and then assign the default based on the instance type. 

and actually is there other method called **copySourceAttributes** where the same attribute is been assigned:
```
if (!description.copySourceCustomBlockDeviceMappings) {
      return description
    } else {
      description.blockDevices = buildBlockDeviceMappingsFromSourceAsg(sourceRegionScopedProvider, sourceAsg, description)
    }
```
is assigned in different places, probably should be convenient to centralize them in other PR.. :information_desk_person:

Reply:
sounds good!